itunes.sql

```diff
@@ -0,0 +1,647 @@
+  INDEX `fk_album_has_user_user1_idx` (`user_id` ASC) VISIBLE,
+  INDEX `fk_album_has_user_album1_idx` (`album_id` ASC) VISIBLE,
+  CONSTRAINT `fk_album_has_user_album1`
+    FOREIGN KEY (`album_id`)
+    REFERENCES `itunes`.`album` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_album_has_user_user1`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `itunes`.`user` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`user_prefer_genre`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`user_prefer_genre` (
+  `user_id` INT NOT NULL,
+  `genre_id` INT NOT NULL,
+  PRIMARY KEY (`user_id`, `genre_id`),
+  INDEX `fk_user_has_genre_genre1_idx` (`genre_id` ASC) VISIBLE,
+  INDEX `fk_user_has_genre_user1_idx` (`user_id` ASC) VISIBLE,
+  CONSTRAINT `fk_user_has_genre_user1`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `itunes`.`user` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_user_has_genre_genre1`
+    FOREIGN KEY (`genre_id`)
+    REFERENCES `itunes`.`genre` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`_playlist_has_song`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`_playlist_has_song` (
+  `user's_playlist_info_id` INT NOT NULL,
+  `song_id` INT NOT NULL,
+  PRIMARY KEY (`user's_playlist_info_id`, `song_id`),
+  INDEX `fk_user's_playlist_info_has_song_song1_idx` (`song_id` ASC) VISIBLE,
+  INDEX `fk_user's_playlist_info_has_song_user's_playlist_info1_idx` (`user's_playlist_info_id` ASC) VISIBLE,
+  CONSTRAINT `fk_user's_playlist_info_has_song_user's_playlist_info1`
+    FOREIGN KEY (`user's_playlist_info_id`)
+    REFERENCES `itunes`.`user's_playlist_info` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_user's_playlist_info_has_song_song1`
+    FOREIGN KEY (`song_id`)
+    REFERENCES `itunes`.`song` (`id`)
+    ON DELETE NO ACTION
+    ON UPDATE NO ACTION)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`song_commercial`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`song_commercial` (
+  `song_id` INT NOT NULL,
+  `price` DECIMAL(2) NOT NULL,
+  `num_of_downloads` INT NOT NULL,
+  PRIMARY KEY (`song_id`),
+  CONSTRAINT `fk_song_commercial_song1`
+    FOREIGN KEY (`song_id`)
+    REFERENCES `itunes`.`song` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`user_credential`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`user_credential` (
+  `user_id` INT NOT NULL,
+  `password` VARCHAR(45) NOT NULL,
```

Comment:
Just so you know, typically you don't store the password in your DB, you store a salt (salt is an encrypted version of password that takes much longer to crack, and when a user logs in, you encrypt the password they provided and compare it with the salt stored) or use an external auth system (e.g. Firebase Auth, AWS Cognito etc.).

This is not required for this task, but just so you know for your future work experience.